tests/test_runner.py

```diff
@@ -0,0 +1,262 @@
+import dataclasses
+from unittest import mock
+
+import pytest
+
+from django_setup_configuration.exceptions import (
+    ConfigurationException,
+    PrerequisiteFailed,
+    SelfTestFailed,
+    ValidateRequirementsFailure,
+    ValidateResultsFailure,
+)
+from django_setup_configuration.runner import (
+    SetupConfigurationRunner,
+    StepExecutionResult,
+)
+from tests.conftest import TestStep
+
+pytestmark = pytest.mark.usefixtures("step_config_env_vars")
+
+
+def test_runner_raises_on_non_existent_step_module_path():
+    with pytest.raises(ConfigurationException):
+        SetupConfigurationRunner(steps=[TestStep, "module.does.not.exist"])
+
+
+def test_runner_raises_on_passing_invalid_step_class():
+    class NotABaseConfigurationStep:
+        pass
+
+    with pytest.raises(ConfigurationException):
+        SetupConfigurationRunner(steps=[TestStep, NotABaseConfigurationStep])
+
+
+@pytest.mark.usefixtures("step_invalid_config_vars")
+def test_validate_requirements_for_step_fails_on_bad_env(
+    runner,
+    runner_step,
+    step_execute_mock,
+    step_validate_results_mock,
+    step_is_configured_mock,
+):
+    with pytest.raises(PrerequisiteFailed):
+        runner._validate_requirements_for_step(runner_step)
+
+    step_is_configured_mock.assert_not_called()
+    step_execute_mock.assert_not_called()
+    step_validate_results_mock.assert_not_called()
+
+
+@pytest.mark.usefixtures("step_invalid_config_vars")
+def test_validate_all_requirements_fails_on_bad_env(
+    runner,
+    step_execute_mock,
+    step_validate_results_mock,
+    step_is_configured_mock,
+):
+    with pytest.raises(ValidateRequirementsFailure) as excinfo:
+        runner.validate_all_requirements()
+
+    assert all(isinstance(exc, PrerequisiteFailed) for exc in excinfo.value.exceptions)
+
+    step_is_configured_mock.assert_not_called()
+    step_execute_mock.assert_not_called()
+    step_validate_results_mock.assert_not_called()
+
+
+@pytest.fixture()
+def expected_successful_run_result(runner_step, expected_step_config):
+    return StepExecutionResult(
```

Comment:
Is this test necessary? Since the test below it probably tests the same initialization but also executes the step?